server: TestClusterVersionMixedVersionTooOld is (very) flaky

```
~/Development/go/src/github.com/cockroachdb/cockroach/pkg master git rev-parse HEAD
8d0c5df6f2e50d8c43b6a0adde7528042fd9620b
~/Development/go/src/github.com/cockroachdb/cockroach/pkg master make stress PKG=./server/ TESTS=TestClusterVersionMixedVersionTooOld
...
--- FAIL: TestClusterVersionMixedVersionTooOld (1.38s)
	version_cluster_test.go:275: 0: incorrect version 1.0 (wanted 1.1)
FAIL


ERROR: exit status 1

3 runs completed, 1 failures, over 1s
```

--- a/pkg/server/version_cluster_test.go
+++ b/pkg/server/version_cluster_test.go
@@ -271,10 +271,10 @@ func TestClusterVersionMixedVersionTooOld(t *testing.T) {
 
 	// Check that we can still talk to the first three nodes.
 	for i := 0; i < tc.NumServers()-1; i++ {
-		if version := tc.getVersionFromSetting(i).Version().MinimumVersion.String(); version != exp {
-			t.Fatalf("%d: incorrect version %s (wanted %s)", i, version, exp)
-		}
 		testutils.SucceedsSoon(tc, func() error {
+			if version := tc.getVersionFromSetting(i).Version().MinimumVersion.String(); version != exp {
+				return errors.Errorf("%d: incorrect version %s (wanted %s)", i, version, exp)
+			}
 			if version := tc.getVersionFromShow(i); version != exp {
 				return errors.Errorf("%d: incorrect version %s (wanted %s)", i, version, exp)
 			}
@@ -325,11 +325,14 @@ func TestClusterVersionMixedVersionTooNew(t *testing.T) {
 
 	// Check that we can still talk to the first three nodes.
 	for i := 0; i < tc.NumServers()-1; i++ {
-		if version := tc.getVersionFromSetting(i).Version().MinimumVersion.String(); version != exp {
-			t.Fatalf("%d: incorrect version %s (wanted %s)", i, version, exp)
-		}
-		if version := tc.getVersionFromShow(i); version != exp {
-			t.Fatalf("%d: incorrect version %s (wanted %s)", i, version, exp)
-		}
+		testutils.SucceedsSoon(tc, func() error {
+			if version := tc.getVersionFromSetting(i).Version().MinimumVersion.String(); version != exp {
+				return errors.Errorf("%d: incorrect version %s (wanted %s)", i, version, exp)
+			}
+			if version := tc.getVersionFromShow(i); version != exp {
+				return errors.Errorf("%d: incorrect version %s (wanted %s)", i, version, exp)
+			}
+			return nil
+		})
 	}
 }

